### PR TITLE
 `HintPrompt` modernization

### PR DESCRIPTION
**Pull request by Erin007**
`HintPrompt` is an old component used in CSF courses to allow students to confirm or cancel a hint request. I made a few code tidying updates to incrementally work towards our [goals of modernizing the frontend](https://docs.google.com/document/d/1Yxek5dqYKLJUraB17yCdEFLlJvlOiB1zhSWZAuu3zPU/edit#heading=h.18w2m61099gz):
- removed the usage of Radium (it's deprecated!) which wasn't actually adding any value to this component any way. Partial for [XTEAM-204](https://codedotorg.atlassian.net/browse/XTEAM-204)
- extracted a couple of inline styles to a SCSS module 
- swapped `LegacyButton` for the DSCO `Button` component. Partial for [XTEAM-430](https://codedotorg.atlassian.net/browse/XTEAM-430)

There are slight visual changes, but no functional differences. 

BEFORE 
hint_prompt_before

AFTER 
hint_prompt_after

### CODE REVIEW

File: apps/src/templates/instructions/HintPrompt.jsx [hunk 1]
```diff
@@ -59,4 +64,4 @@ HintPrompt.propTypes = {
   textToSpeechEnabled: PropTypes.bool,
 };
 
-export default Radium(HintPrompt);
+export default HintPrompt;
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/HintPrompt.jsx:67]: 🤩

File: apps/src/templates/instructions/hint-prompt.module.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.button {
+  border: 2px solid black;
+  &Yes {
+    margin-right: 5px;
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/hint-prompt.module.scss]: Just double checking if this also works in RTL?

**Review by sanchitmalhotra126 — APPROVED** [review]
🎉

File: apps/src/templates/instructions/hint-prompt.module.scss (hunk 2 above)
Comment (kelbyhawn) [reply apps/src/templates/instructions/hint-prompt.module.scss]: If it's not try `margin-inline-end: 5px;` — you could also update the margin here to `8px` (if it looks ok) to match the grid 🙌

Comment (Erin007) [reply apps/src/templates/instructions/hint-prompt.module.scss]: Great catch! The buttons were squished on prod in RTL so this little improvement is extra icing for RTL language users 🎉 

BEFORE on production 
Screenshot 2024-08-22 at 11 36 43 AM

AFTER on local 
Screenshot 2024-08-22 at 11 48 20 AM

**Pull request merged**